lib/screens/topic_screen.dart

```diff
@@ -37,41 +38,51 @@ class _TopicScreenState extends State<TopicScreen> {
     });
   }
 
-  List<Widget> _createPageViewWidgets(BuildContext context) {
-    for (var i = 0; i < _articles.length; i++) {
-      _pageViewWidgets.add(new ArticlePage(
-        topicId: _articles[i].topicId,
-        articleId: _articles[i].id,
-        name: _articles[i].name,
-        text: _articles[i].text,
-        audio: _articles[i].audio,
-        video: _articles[i].video,
-        image: _articles[i].image,
-        order: _articles[i].order,
-      ));
-    }
-    return _pageViewWidgets;
-  }
-
   @override
   void initState() {
-    // TODO: implement initState
     super.initState();
     _initTopic();
   }
 
-  _forwardButtonBehaviour() {
+  void _forwardButtonBehaviour() {
     pageController.nextPage(
         duration: new Duration(milliseconds: 500), curve: Curves.fastOutSlowIn);
+    if (pageController.page == 0.0) {
+      setState(() {
+        _isBackwardDisable = false;
+      });
+    }
+    if (pageController.page + 1.0 == (_articles.length - 1).toDouble()) {
+      setState(() {
+        _isForwardDisable = true;
+      });
+    }
   }
 
-  _backwardButtonBehaviour() {
+  void _backwardButtonBehaviour() {
     pageController.previousPage(
         duration: new Duration(milliseconds: 500), curve: Curves.fastOutSlowIn);
+    if (pageController.page == (_articles.length - 1).toDouble()) {
+      setState(() {
+        _isForwardDisable = false;
+      });
+    }
+    if (pageController.page - 1.0 == 0.0) {
```

Comment:
Do not introduce double arithmetic into an int, especially when using it to compare.